src/main/java/org/datavyu/plugins/ffmpeg/MpvMediaPlayer.java

```diff
@@ -1,153 +1,63 @@
 package org.datavyu.plugins.ffmpeg;
 
-//import sun.awt.windows.WComponentPeer;
-import com.sun.javafx.tk.TKStage;
-import javafx.stage.Stage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.datavyu.util.NativeLibraryLoader;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
-
-
-import javax.swing.*;
-import java.awt.*;
-import java.lang.reflect.Method;
 import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
 
 
-public class MpvMediaPlayer extends FfmpegMediaPlayer{
+public abstract class MpvMediaPlayer extends FfmpegMediaPlayer {
+    /**
+     * Holds the library name and version
+     */
+    private final static class LibraryDependency {
+        LibraryDependency(String name, String version) {
+            this.name = name;
+            this.version = version;
+        }
+        public String getFullName() {
+            return name + SEPARATOR + version;
+        }
+        private static final char SEPARATOR = NativeLibraryLoader.isMacOs ? '.' : '-';
+        private String name;
+        private String version;
+    }
+
+    private static final Logger LOGGER = LogManager.getFormatterLogger(MpvMediaPlayer.class);
+    private static final List<LibraryDependency> librariesToExtract = new ArrayList<LibraryDependency>() {{
```

Comment:
Please rename this to mpvLibrariesToExtract and move to a util class along with ffmpegLibrariesToextract and sdlLibrariesToExtract for the Java and SDL media player, respectively.

Reply:
I've added the extract method to the existing NativeLibraryLoader along with the LibraryDependency class. However, I left the listings in the Ffmpeg/MPV classes. Rational: If want to remove players later it is better to keep these dependencies separate.

Reply:
Totally agree with that